### PR TITLE
feat: publish the OLM bundles in the artifacts directory

### LINKED ISSUE #4038: [Feature]: Publish OLM bundles in the artifacts repo
### Is there an existing issue already for this feature request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new feature request to be evaluated.

### What problem is this feature going to solve? Why should it be added?

Currently, the only way to retrieve the bundle that its published on OperatorHub.io is using the one pushed there

### Describe the solution you'd like

We should publish this as our own artifact too, this will help us to keep track of the versions and make easy to users to find the bundle if they need it

### Describe alternatives you've considered

None

### Additional context

_No response_

### Backport?

No

### Are you willing to actively contribute to this feature?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sxd**
This will allow any user to use the bundle easily without waiting
for this to be published in OperatorHub.io

Closes #4038

### CODE REVIEW

**Review by jsilvela — APPROVED** [review]

**Review by jlong49 — APPROVED** [review]

**Comment by jlong49** [comment]
@cloudnative-pg/maintainers can I get an approval on this please?

**Review by mnencia — APPROVED** [review]

**Pull request merged**
